pkg/client/cache/fifo.go

```diff
@@ -241,6 +241,29 @@ func (f *FIFO) Replace(list []interface{}, resourceVersion string) error {
 	return nil
 }
 
+// Resync will touch all objects to put them into the processing queue
+func (f *FIFO) Resync() error {
+	f.lock.Lock()
+	defer f.lock.Unlock()
+
+	for id := range f.items {
+		found := false
+		for _, qid := range f.queue {
```

Comment:
Sorry - I missed this before.
This is basically O(|items|*|queue|) which may be big.

Can you please first create a set of ids from f.queue and then just lookup in this set?